File: 83 Remove Duplicates from Sorted List/deleteDuplicates.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        record = set()
        cur = head
        pre = None
        while cur != None:
            if cur.val in record:
                pre.next = cur.next
                cur = cur.next
            else:
                record.add(cur.val)
                pre = cur
                cur = cur.next
        return head

File: 83 Remove Duplicates from Sorted List/deleteDuplicates.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        
        Set<Integer> record = new HashSet();
        
        ListNode cur = head;
        ListNode pre = null;
        while( cur != null ){
            if( record.contains(cur.val) ){
                pre.next = cur.next;
                cur = cur.next;
            }
            else{
                record.add(cur.val);
                pre = cur;
                cur = cur.next;
            }
        }
        return head;
    }
}